src/logion/migration/1695136615896-AddLinkSubmitterAndLifecycle.ts

```diff
@@ -0,0 +1,99 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddLinkSubmitterAndLifecycle1695136615896 implements MigrationInterface {
+    name = 'AddLinkSubmitterAndLifecycle1695136615896'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+
+        await this.addSubmitter(queryRunner);
+        await this.addLifecycle(queryRunner);
+    }
+    
+    async addSubmitter(queryRunner: QueryRunner): Promise<void> {
+        // submitter
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "submitter_address" character varying(255)`);
+        await queryRunner.query(`
+            UPDATE "loc_link" item
+            SET "submitter_address" = (SELECT owner_address FROM loc_request request where item.request_id = request.id )
+        `);
+        await queryRunner.query(`ALTER TABLE "loc_link" ALTER COLUMN "submitter_address" SET NOT NULL`);
+
+        // submitter_address_type
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "submitter_address_type" character varying(255)`);
+        await queryRunner.query(`UPDATE "loc_link" SET "submitter_address_type" = 'Polkadot'`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ALTER COLUMN "submitter_address_type" SET NOT NULL`);
+    }
+
+    async addLifecycle(queryRunner: QueryRunner): Promise<void> {
+
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "status" character varying(255)`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "reject_reason" character varying(255)`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "reviewed_on" TIMESTAMP`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "acknowledged_by_owner_on" TIMESTAMP`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "acknowledged_by_owner" boolean NOT NULL DEFAULT false`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "acknowledged_by_verified_issuer_on" TIMESTAMP`);
+        await queryRunner.query(`ALTER TABLE "loc_link" ADD "acknowledged_by_verified_issuer" boolean NOT NULL DEFAULT false`);
+
+        // draft=false > ACKNOWLEDGED
+        await queryRunner.query(`
+            UPDATE "loc_link" item
+            SET "status" = 'ACKNOWLEDGED', "acknowledged_by_owner" = true
+            WHERE "draft" = 'false'
+        `);
+
+        // LOC DRAFT => ITEM draft=true > DRAFT
+        await queryRunner.query(`
+            UPDATE "loc_link" item
+            SET "status" = 'DRAFT'
+            FROM loc_request loc
+            WHERE 1=1
+              AND item.draft = 'true'
+              AND loc.id = item.request_id
+              AND loc.status = 'DRAFT'
+        `);
+        // LOC (REQUESTED, OPEN), ITEM draft=true > REVIEW_ACCEPTED
```

Comment:
I am not sure this and the following update rules make sense as all links were submitted by the owner up to now and therefore, no review is needed.